client/src/components/Layout/index.tsx

```diff
@@ -40,12 +71,29 @@ export function Header() {
         </Link>
 
         <NavigationBar />
-        <a
-          className="text-white hover:cursor-pointer hover:underline"
-          href="/api/logout"
-        >
-          Logout
-        </a>
+
+        {/* User dropdown */}
+        <div className="relative" ref={menuRef}>
+          <button
+            type="button"
+            onClick={() => setMenuOpen((prev) => !prev)}
+            className="font-public-sans hover:bg-blue-cool-70 flex items-center gap-2 rounded px-3 py-2 text-white focus:outline-none"
+          >
+            <Icon.Person size={3} aria-hidden />
+            {currentUser ? currentUser.username : 'Loading...'}
```

Comment:
This will no longer need to load with the other changes mentioned.
```suggestion
            {user.username}
```